Source/WebCore/page/scrolling/ScrollAnchoringController.h

```diff
@@ -41,12 +41,18 @@ class ScrollAnchoringController final {
     void invalidateAnchorElement();
     void updateScrollPosition();
     void selectAnchorElement();
+    void chooseAnchorElement();
+    void updateAnchorElement();
+    void printAnchorElement();
     LocalFrameView& frameView() { return m_frameView; }
 
 private:
     LocalFrameView& m_frameView;
+    ScrollOffset computeOffset(RenderBox& candidate);
     WeakPtr<Element, WeakPtrImplWithEventTargetData> m_anchorElement;
-    FloatPoint m_lastPositionForAnchorElement;
+    ScrollOffset m_lastOffsetForAnchorElement;
+    bool m_midUpdatingScrollPositionForAnchorElement { false };
+    bool m_isQueuedForScrollPositionUpdate { false };
```

Comment:
There's some state aliased between `m_isQueuedForScrollPositionUpdate` and a `ScrollAnchoringController` being a member of the `m_scrollableAreasWithScrollAnchoringControllersNeedingUpdate` set. Try to avoid having both.

Reply:
I'm not sure there's a great way to get rid of m_isQueuedForScrollPositionUpdate because we have no way of checking if we were queued in `ScrollAnchoringController::adjustScrollPositionForAnchoring` (since the scrollable area is popped by then). However in the future if ScrollAnchoringController::adjustScrollPositionForAnchoring is only called on the queued scrollable areas it might be ok to remove m_isQueuedForScrollPositionUpdate and not check in ScrollAnchoringController::adjustScrollPositionForAnchoring.